lib/event-handlers.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+module.exports = function (job, collection) {
+  const log = require('./log')(job.meta.name);
+  const later = require('later');
+
+  return {
+    onTickStarted: () => {
+      log.info('tick started for %s', job.meta.name);
+
+      // get the next run time
+      let nextSchedule = later.parse.cron(job._opts.cronParams.cronTime);
+      nextSchedule = later.schedule(nextSchedule).next(2);
+      nextSchedule = nextSchedule[nextSchedule.length - 1];
+
+      const data = {
+        name: job.meta.name,
+        startTime: new Date(),
+        finishTime: null,
+        executionDuration: null,
+        lastStatus: 'In Process',
+        nextSchedule: nextSchedule
+      };
+
+      return collection.then((coll) => coll.insert(data))
+        .then((res) => {
+          // Storing the mongo in job's meta data for doc update on complete
+          job.meta.id = res.ops[0]._id;
+          log.info(`meta for job ${job.meta.name} %j`, job.meta);
+        })
+        .catch((e) => {
+          // Delete an older _id if it exists since we don't want the
+          // TICK_COMPLETE to update the old/wrong entry
+          delete job.meta.id;
```

Comment:
So the reason I did this was because when reading I realised that on the off chance that this write fails `TICK_COMPLETE` could update an older record using an outdated `_id`